### PR TITLE
packaging: add scripts for debian

### PR DESCRIPTION
**Pull request by KKoukiou**
Copyright and changelog are missing, to be added when this is ready for packaging in debian.

### CODE REVIEW

**Comment by martinpitt** [comment]
I fixed the project's webhook. Note that it's most likely going to fail as you *have* to have a debian/changelog and debian/copyright to build a source package. Just take/adapt cockpit-podman's.

**Comment by martinpitt** [comment]
dpkg-parsechangelog: error: cannot open file debian/changelog: No such file or directory

Yup :grin:

**Comment by jelly** [comment]
I've taken the liberty of rebasing and re-doing the existing users commit (was easier to redo due to conflicts).

**Comment by jelly** [comment]
TestNavigator.testHiddenItems - seems flaky

@martinpitt mind checking if at least the packaging changes are ok?

File: packaging/debian/control [hunk 1]
```diff
@@ -0,0 +1,24 @@
+Source: cockpit-navigator
+Section: admin
+Priority: optional
+Maintainer: Utopia Maintenance Team <pkg-utopia-maintainers@lists.alioth.debian.org>
+Uploaders: Martin Pitt <mpitt@debian.org>,
+Build-Depends: debhelper-compat (= 13),
+               gettext (>= 0.19.7),
+               gettext (>= 0.21) | appstream,
```
Comment (jelly) [inline packaging/debian/control:8]: compared to podman, `gettext` is not included, nor `appsteam`

File: packaging/debian/control [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+Rules-Requires-Root: no
+Homepage: https://github.com/cockpit-project/cockpit-navigator
+Vcs-Git: https://salsa.debian.org/debian/cockpit-navigator.git
+Vcs-Browser: https://salsa.debian.org/debian/cockpit-navigator
+
+Package: cockpit-navigator
+Architecture: all
+Multi-Arch: foreign
+Depends: ${misc:Depends},
+         cockpit-bridge (>= 294),
```
Comment (jelly) [inline packaging/debian/control:19]: this might change in the future depending on `fslist1`  improvements

File: packaging/debian/rules [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+override_dh_auto_clean:
+	# don't call `make clean`, in a release dist/ is precious
+	rm -f po/LINGUAS
+
+override_dh_auto_test:
+	# don't call `make check`, these are integration tests
+
+override_dh_auto_install:
+	make install DESTDIR=debian/cockpit-navigator PREFIX=/usr
```
Comment (jelly) [inline packaging/debian/rules:14]: podman does not override `auto_install` interestingly just exports PREFIX=/usr

File: packaging/debian/control (hunk 1 above)
Comment (martinpitt) [reply packaging/debian/control:8]: But c-navigator already has a metadata file. However, all supported Debian and Ubuntu releases now have gettext ≥ 0.21, so these two lines should change to `gettext (>= 0.21)` (we can do the same on our other projects).

File: packaging/debian/control [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+Maintainer: Utopia Maintenance Team <pkg-utopia-maintainers@lists.alioth.debian.org>
+Uploaders: Martin Pitt <mpitt@debian.org>,
+Build-Depends: debhelper-compat (= 13),
+               gettext (>= 0.19.7),
+               gettext (>= 0.21) | appstream,
+Standards-Version: 4.6.2
+Rules-Requires-Root: no
+Homepage: https://github.com/cockpit-project/cockpit-navigator
+Vcs-Git: https://salsa.debian.org/debian/cockpit-navigator.git
+Vcs-Browser: https://salsa.debian.org/debian/cockpit-navigator
```
Comment (martinpitt) [inline packaging/debian/control:13]: These don't exist yet, but indeed that's where they will be, so let's keep them in.

File: packaging/debian/copyright [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+ with node, npm, and a bundler. node_modules/ is not shipped as part of the
+ upstream release tarballs, but can be reconstructed precisely through the
+ shipped package-lock.json with the command "npm install".  Rebuilding files in
+ dist/ requires internet access as that process needs to download additional
+ npm modules from the Internet, thus upstream ships the pre-minified bundles
+ as part of the upstream release tarball so that the package can be built
+ without internet access and lots of extra unpackaged build dependencies.
+
+Files: *
+Copyright: 2016-2020  Red Hat, Inc.
```
Comment (martinpitt) [inline packaging/debian/copyright:16]: These years are most definitively wrong. Possibly already in the upstream file headers.

File: packaging/debian/copyright [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+ Lesser General Public License for more details.
+ .
+ You should have received a copy of the GNU General Public License
+ along with this program. If not, see <https://www.gnu.org/licenses/>.
+ .
+ On Debian systems, the complete text of the GNU Lesser General
+ Public License can be found in "/usr/share/common-licenses/LGPL-2.1".
+
+Files: *.metainfo.xml
+Copyright: Copyright (C) 2018 Red Hat, Inc.
```
Comment (martinpitt) [inline packaging/debian/copyright:35]: Dito

File: packaging/debian/rules (hunk 3 above)
Comment (martinpitt) [reply packaging/debian/rules:14]: Yess, please use the same d/rules file as current podman.

File: packaging/debian/changelog.in [hunk 7]
```diff
@@ -1 +1,5 @@
 cockpit-navigator (VERSION-1) UNRELEASED; urgency=medium
+
+  * Upstream build
+
+ -- Cockpit <cockpit-devel@lists.fedorahosted.org>  Wed, 27 Sep 2023 13:34:21 +0200
```
Comment (martinpitt) [inline packaging/debian/changelog.in:5]: This belongs into the first commit, not "Add Makefile rule to generate debian changelog". I.e. rebase error.

File: packaging/debian/control [hunk 8]
```diff
@@ -16,10 +16,9 @@ Package: cockpit-navigator
 Architecture: all
 Multi-Arch: foreign
 Depends: ${misc:Depends},
-         cockpit-bridge (>= 294)
+         cockpit-bridge (>= 294),
```
Comment (martinpitt) [inline packaging/debian/control]: Adding that trailing comma also belongs into the first commit.

File: packaging/debian/control [hunk 9]
```diff
@@ -16,10 +16,9 @@ Package: cockpit-navigator
...
 Description: A filesystem browser for Cockpit
-
  The Cockpit Web Console enables users to administer GNU/Linux servers using a
  web browser.
-
+ .
  This package adds an user interface for browsing the filesystem.
```
Comment (martinpitt) [inline packaging/debian/control]: And this noise, too.

File: test/check-application [hunk 10]
```diff
@@ -536,17 +538,26 @@ class TestNavigator(testlib.MachineCase):
...
+        if any(image in m.image for image in DISTROS_WITH_SELINUX):
+            selinux_dot = "."
+        else:
+            selinux_dot = ""
+
+        def ls_output(filename):
+            output = m.execute(f"ls -l {filename}")[0:23]
+            return output.strip()
+
+        self.assertEqual(ls_output("/home/admin/test-sym"), f"lrwxrwxrwx{selinux_dot} 1 root root")
```
Comment (martinpitt) [inline test/check-application:551]: That `DISTROS_WITH_SELINUX` is rather clunky -- why not `assertRegexp` with `"lrwxrwxrwx.* 1 root root"`?

**Review by martinpitt — CHANGES_REQUESTED** [review]
Thanks! There's a few small cleanups to do, but otherwise looks fine. @jelly do you want to do them, or want me to?

File: test/check-application (hunk 10 above)
Comment (jelly) [reply test/check-application:551]: Agreed, nice suggestion!

File: packaging/debian/rules (hunk 3 above)
Comment (jelly) [reply packaging/debian/rules:14]: Ack, already applied this locally works as expected

File: packaging/debian/changelog.in (hunk 7 above)
Comment (jelly) [reply packaging/debian/changelog.in:5]: Uh hmm, I don't think that can be a rebase issue but shall I just squash it to a generic add packaging commit.

**Comment by martinpitt** [comment]
@jelly I suppose you forgot to repush?

**Comment by jelly** [comment]
> @jelly I suppose you forgot to repush?

Yes, sorry. Cruise control was pushing to my own origin and not Katerina's!

**Review by martinpitt — APPROVED** [review]
Thanks!

**Comment by jelly** [comment]
Follow up for bots: https://github.com/cockpit-project/bots/pull/5781

**Pull request merged**